src/main/java/core/basesyntax/service/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storageFruits.put(transaction.getFruit(), storageFruits
+                .getOrDefault(transaction.getFruit(), 0) - transaction.getQuantity());
```

Comment:
Before saving the result balance in the storage, you must check that the resulting balance is positive. If the balance is negative, throw a `RuntimeException` as per the checklist requirement for handling the Purchase operation.